Kentico.Kontent.Management/Models/Shared/Reference.cs

```diff
@@ -45,4 +46,58 @@ public sealed class Reference
     /// </summary>
     /// <param name="externalId">The external id of the identifier.</param>
     public static Reference ByExternalId(string externalId) => new() { ExternalId = externalId };
+
+    /// <summary>
+    /// Transforms the dynamic object to the <see cref="Reference"/>
+    /// </summary>
+    public static Reference FromDynamic(dynamic source)
+    {
+        if (DynamicExtensions.HasProperty(source, "id"))
```

Comment:
Again, we should be more defensive and put this whole FromDynamic into try-catch block and throw a descriptive exception if the Reference object cannot be created.